salt/states/archive.py

```diff
@@ -1220,6 +1238,26 @@ def extracted(name,
             _add_explanation(ret, source_hash_trigger, contents_missing)
             return ret
 
+        if clean_top and contents is not None:
+            errors = []
+            log.debug('Removing directory %s due to clean_top set to True', name)
+            try:
+                salt.utils.files.rm_rf(name.rstrip(os.sep))
```

Comment:
Maybe rename the directory, if the archive.extract is successful, then delete, otherwise roll it back

Reply:
I'm not sure here.
First of all, it adds complexity + a hidden requirement for a 2x space(to keep both old and new dir at some point of time).
Second is that the current `clean: True` logic doesn't do it and I feel like it's more or less the same case.